gpu_support/cuda_utils/install_cuda_compatlibs_loop.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+cuda_install_dir=$1
+install_cuda_version=$2
+
+# Check if the CUDA compat libraries are installed and compatible with the target CUDA version
+# if not find the latest version of the compatibility libraries and install them
+
+# get URL to latest CUDA compat libs, exit if URL is invalid
+cuda_compat_urls="$($(dirname "$BASH_SOURCE")/get_cuda_compatlibs.sh)"
+ret=$?
+if [ $ret -ne 0 ]; then
+  echo "Couldn't find current URLs of the CUDA compat libraries, instead got:"
+  echo $cuda_compat_urls
+  exit 1
+fi
+
+# loop over the compat library versions until we get one that works for us
+keep_driver_check=1
+# Do a maximum of five attempts
+for value in {1..5}
```

Comment:
We can do better here. We should probably keep going until we hit a compat version that won't satisfy our `install_cuda_version` requirement. Using `/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/latest/eessi_compat_version.txt.tmp` and `/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia/latest/eessi_compat_version.txt` should be able to help us here.